ci-operator/step-registry/cucushift/e2e/cucushift-e2e-commands.sh

```diff
@@ -40,12 +40,44 @@ function echo_e2e_tags() {
 }
 function filter_test_by_version() {
     local xversion yversion
-    IFS='.' read xversion yversion _ < <(oc version -o yaml | yq '.openshiftVersion')
+    OPENSHIFT_VERSION="$(oc version -o yaml | yq '.openshiftVersion')"
+    IFS='.' read xversion yversion _ <<< "$OPENSHIFT_VERSION"
     if [[ -n $xversion ]] && [[ $xversion -eq 4 ]] && [[ -n $yversion ]] && [[ $yversion =~ [12][0-9] ]] ; then
         export E2E_RUN_TAGS="${E2E_RUN_TAGS} and @${xversion}.${yversion}"
     fi
     echo_e2e_tags
 }
+function filter_test_by_arch() {
+    local version node_archs
+    version="$OPENSHIFT_VERSION"
+    if [[ -n "$version" ]] ; then
+        if [[ "$version" =~ multi ]] ; then
+            mapfile -t node_archs < <(oc get nodes -o yaml | yq '.items[].status.nodeInfo.architecture' | sort -u)
+            case "${#node_archs[@]}" in
+                1)
+                    export E2E_RUN_TAGS="${E2E_RUN_TAGS} and @${node_archs[0]}"
+                    ;;
+                2)
+                    export E2E_RUN_TAGS="${E2E_RUN_TAGS} and @heterogeneous and @${node_archs[0]} and @${node_archs[1]}"
+                    ;;
+                *)
+                    echo "=============================================="
+                    echo "Error: got unexpected arch, '${node_archs[*]}'"
+                    echo "=============================================="
+                    ;;
+            esac
+        elif [[ "$version" =~ (aarch64|arm64) ]] ; then
+            export E2E_RUN_TAGS="${E2E_RUN_TAGS} and @arm64"
+        elif [[ "$version" =~ ppc64le ]] ; then
+            export E2E_RUN_TAGS="${E2E_RUN_TAGS} and @ppc64le"
+        elif [[ "$version" =~ s390x ]] ; then
+            export E2E_RUN_TAGS="${E2E_RUN_TAGS} and @s390x"
+        else
+            export E2E_RUN_TAGS="${E2E_RUN_TAGS} and @amd64"
+        fi
+    fi
+    echo_e2e_tags
+}
```

Comment:
```suggestion
function filter_test_by_arch() {
    local node_archs # << is this needed ??
    mapfile -t node_archs < <(oc get nodes -o yaml | yq '.items[].status.nodeInfo.architecture' | sort -u)
    case "${#node_archs[@]}" in
        1)
            export E2E_RUN_TAGS="${E2E_RUN_TAGS} and @${node_archs[0]}"
            ;;
        2)
            export E2E_RUN_TAGS="${E2E_RUN_TAGS} and @heterogeneous and @${node_archs[0]} and @${node_archs[1]}"
            ;;
        *)
            echo "=============================================="
            echo "Error: got unexpected arch, '${node_archs[*]}'"
            echo "=============================================="
            ;;
    esac
    echo_e2e_tags
}
```

We don't need to check the version either as it's not a good way to catch the arch of the cluster. Just checking the nodes arch is what we are doing around.

Reply:
Looking at it now, I still think we should approach the problem once and for all with the proper solution and definitive approach:


```suggestion
function join_by_prefix() {
  local delimiter="${1}"
  shift
  local array=("$@")
  local result=""
  local i
  for ((i=0; i<${#array[@]}; i++)); do
    result+="${delimiter}${array[i]}"
  done
  echo -n "$result"
}

function filter_test_by_arch() {
    local node_archs # << is this needed ??
    mapfile -t node_archs < <(oc get nodes -o yaml | yq '.items[].status.nodeInfo.architecture' | sort -u)
    export E2E_RUN_TAGS="${E2E_RUN_TAGS}$(join_by_prefix " and @" "${node_archs[@]}")"
    if [ "${#node_archs[@]}" -gt 1 ]; then
      export E2E_RUN_TAGS="${E2E_RUN_TAGS} and @heterogeneous"
    fi
    echo_e2e_tags
}
```

Reply:
@aleskandro The latest version should also works for the more than 2 architectures